zerver/views/message_edit.py

```diff
@@ -93,6 +94,30 @@ def fill_edit_history_entries(
     return formatted_edit_history
 
 
+def is_resolve_topic_request(
+    user_profile: UserProfile,
+    message_id: PathOnly[NonNegativeInt],
+    new_topic_name: OptionalTopic,
+    propagate_mode: Literal["change_later", "change_one", "change_all"],
+    content: Optional[str],
+) -> bool:
+    message = access_message(user_profile, message_id)
+    old_topic_name = message.topic_name()
+
+    if content is None and new_topic_name is not None and propagate_mode == "change_all":
+        topic_resolved: bool = new_topic_name.startswith(
+            RESOLVED_TOPIC_PREFIX
+        ) and not old_topic_name.startswith(RESOLVED_TOPIC_PREFIX)
+        topic_unresolved: bool = old_topic_name.startswith(
+            RESOLVED_TOPIC_PREFIX
+        ) and not new_topic_name.startswith(RESOLVED_TOPIC_PREFIX)
+
+        if topic_resolved or topic_unresolved:
+            return True
```

Comment:
I believe the conditions are enough for determining if it is a resolve topic request?